tiny_dnn/network.h

```diff
@@ -601,6 +601,7 @@ class network {
     const_iterator begin() const { return net_.begin(); }
     const_iterator end() const { return net_.end(); }
 
+#ifndef CNN_NO_SERIALIZATION
```

Comment:
Throwing an exception or `assert(false);` might be good in case `#ifdef CNN_NO_SERIALIZATION`